Doppler.BillingUser/Controllers/BillingController.cs

```diff
@@ -255,6 +255,44 @@ public async Task<IActionResult> UpdateCurrentPaymentMethod(string accountname,
             }
         }
 
+        [Authorize(Policies.PROVISORY_USER_OR_SUPER_USER)]
+        [HttpGet("/accounts/{accountname}/declined-invoices")]
+        public async Task<IActionResult> GetDeclinedInvoices(string accountname)
+        {
+            var userBillingInfo = await _userRepository.GetUserBillingInformation(accountname);
+
+            if (userBillingInfo.PaymentMethod != PaymentMethodEnum.CC && userBillingInfo.PaymentMethod != PaymentMethodEnum.MP)
+            {
+                return new BadRequestObjectResult("Payment method not supported")
+                {
+                    StatusCode = 500
+                };
+            }
+
+            var user = await _userRepository.GetUserInformation(accountname);
+
+            var invoices = await _billingRepository.GetDeclinedInvoices(user.IdUser);
+
+            var declinedInvoicesData = new List<DeclinedInvoiceData>();
+            var total = 0.0M;
+            foreach (var invoice in invoices)
+            {
+                total += invoice.Amount;
+                declinedInvoicesData.Add(new DeclinedInvoiceData()
+                {
+                    Date = invoice.Date,
+                    InvoiceNumber = invoice.InvoiceNumber,
+                    Amount = invoice.Amount
+                });
+            }
+
+            return new OkObjectResult(new GetDeclinedInvoicesResult()
+            {
+                TotalPending = total,
+                Invoices = declinedInvoicesData
+            });
```

Comment:
In general, when it is possible, using declarative Linq code is better than imperative loop mutation a collection. 

For example:

```suggestion
            var declinedInvoicesData = invoices.Select(invoice => new DeclinedInvoiceData()
            {
                Date = invoice.Date,
                InvoiceNumber = invoice.InvoiceNumber,
                Amount = invoice.Amount
            })
            .ToList();

            return new OkObjectResult(new GetDeclinedInvoicesResult()
            {
                TotalPending = declinedInvoicesData.Sum(x => x.Amount),
                Invoices = declinedInvoicesData
            });
```

Reply:
Done!